### ISSUE TITLE
Remove deprecated things - final check

### ISSUE THREAD

**Issue by mpg** [ARMmbed/mbedtls#4285]
We have a number of issues that, together, are supposed to remove everything that had been deprecated from the code base:

- #4031 Remove obsolete TLS features **(closed/merged)**
- #4278 Remove 'mode' parameter from RSA functions **(closed/merged)**
- #4078 Remove PKCS11 module **(closed/merged)**
- #4280 Remove deprecated functions from hashing modules **(closed/merged)**
- #4283 Remove deprecated error codes **(closed/merged)**
- #4284 Remove deprecated types and constants in PSA **(closed/merged)**
- #4282 Remove deprecated things - misc/remainder **(closed/merged)**

The present ticket should be done after all of these has been closed, and consist of a final check that nothing deprecated remains in the library. This check can consist in running `git grep -i deprecated` and checking that all the occurrences found are justifiable (documentation, definition of the deprecation macros, tests, etc.)

If any unwanted occurrence is found, a PR should be raised to remove it; otherwise, this issue can be closed without a PR.

**Comment by mpg** [comment]
I went through the ouput of `grep -Ri deprecated library include` and sorted the matches as follows:

False positives
---------------

`library/CMakeLists.txt` using `-Wno-documentation-deprecated-sync` is good.

`include/mbedtls/platform_util.h`, `include/mbedtls/config.h` and `include/mbedtls/check_config.h`: document and implement `MBEDTLS_DEPRECATED_WARNING` and `MBEDTLS_DEPRECATED_REMOVED` which should stay even if they're currently unused.

`include/psa/crypto_compat.h`: talking about things that _will be_ deprecated. Probably OK.

`include/psa/crypto_values.h`: calling (3)DES and RC4 deprecated, but more in a moral sense than as in "we're dropping them from the API". We're dropping RC4, see #4084, and are considering dropping single-DES and 2-key DES, see #4396, but that doesn't prevent people from providing drivers implementing them so the constants defined here definitely should stay.

True positives
--------------

`include/mbedtls/ssl.h` defining `MBEDTLS_SSL_MINOR_VERSION_1` and `MBEDTLS_SSL_MINOR_VERSION_2` - those should no longer be part of the public API, but are still used in the code; removing them entirely is tracked is part of #4564 which is a 3.x task. In the meantime, I suggest we move the definitions out of the public API and into a private header.

`include/mbedtls/rsa.h` a note says that direct manipulation of the fields of the context structure is deprecated - suggesting to remove that note as part of #4511

`include/mbedtls/rsa.h`: "It is deprecated and a security risk to not provide a PRNG" - tracked as part of #4544

`include/mbedtls/ecdsa.h`: "[not providing a PRNG] is DEPRECATED." - also tracked as part of #4544

**Comment by gilles-peskine-arm** [comment]
I did the same analysis on 84191eab064ef103f72e461182e6f7ca73e7c448. I also looked in the root files and inside `programs` and didn't find a mention of deprecation by name. I also ran `agrep -2 -i deprecated` in case we'd misspelled “deprecated” and it found nothing else that was relevant. The only potentially relevant hits were in `include/mbedtls`.

* `config.h`: just mentioning the possibility of deprecation
* `ecdsa.h`: absence of RNG for blinding → #4544
* `platform_util.h`: just mentioning the possibility of deprecation
* `rsa.h`: direct manipulation of the context → https://github.com/ARMmbed/mbedtls/pull/4511#discussion_r645425518
* `rsa.h`: absence of RNG for blinding → #4544
* `ssl.h`: a small holdover of the partial removal of TLS 1.[01] → https://github.com/ARMmbed/mbedtls/pull/4616

I agree with your analysis and it looks complex to me.
